src/index.js

```diff
@@ -462,7 +462,7 @@ export var h = function(name, props) {
   }
 
   return typeof name === "function"
-    ? name(props, (props.children = children))
+    ? name(props, name !== Lazy && (props.children = children))
```

Comment:
This is not the only problem with a current particular line of code.
For example your immutable state may become mutable if you will use a spread operator like this:
```js
const stateOrSlice = { a: 1 } // immutable state will be mutated by view
<div {...stateOrSlice} />     // dangerous usage
console.log(stateOrSlice)     // oh no => { a: 1, children: [] }
```
A discussion about the problem is here: https://github.com/jorgebucaran/hyperapp/pull/726#discussion_r201326488

Reply:
@frenzzy Do you know how other frameworks solve this problem?

Reply:
by copying instead of mutation? https://github.com/facebook/react/blob/v16.8.3/packages/react/src/ReactElement.js#L175

Reply:
@frenzzy We should do this right. Want to send me a PR? :)

Reply:
and after it they have to use shallow compare instead of `===` for React.memo =/

Reply:
here we go: https://github.com/jorgebucaran/hyperapp/pull/813